dependency-convergence/src/main/java/com/google/cloud/cloudbomcommitcheck/ArtifactData.java

```diff
@@ -0,0 +1,234 @@
+      } else {
+        lastUpdated = "";
+      }
+      return new LatestMetadata(latestVersion, lastUpdated);
+    } catch (XmlPullParserException | ParseException | IOException ignored) {
+      return new LatestMetadata("", "");
+    }
+  }
+
+  /**
+   * Grabs the Github URL from the POM's scm section. If the URL is not found, then this checks the
+   * associated parent POM (given that the current POM is not already the parent of our artifact)
+   *
+   * @param pomUrl the Maven URL of the artifact's POM file
+   * @return artifact's Github URL
+   */
+  private static String getScmGithubUrl(String pomUrl) {
+    try {
+      File pomFile = File.createTempFile("pomFile", ".xml");
+      pomFile.deleteOnExit();
+      URL url = new URL(pomUrl);
+      ByteStreams.copy(url.openStream(), new FileOutputStream(pomFile));
+      MavenXpp3Reader read = new MavenXpp3Reader();
+      Model model = read.read(new FileInputStream(pomFile));
+      if (model == null) {
+        return null;
+      }
+      return model.getUrl();
+    } catch (XmlPullParserException | IOException ignored) {
+      return null;
+    }
+  }
+
+  private static String generatePomFileUrl(Artifact artifact) {
+    String groupPath = artifact.getGroupId().replace('.', '/');
+    return basePath + "/" + groupPath
+        + "/" + artifact.getArtifactId()
+        + "/" + artifact.getVersion()
+        + "/" + artifact.getArtifactId() + "-" + artifact.getVersion() + ".pom";
+  }
+
+  private static String generateMetadataUrl(Artifact artifact) {
+    String groupPath = artifact.getGroupId().replace('.', '/');
+    return basePath + "/" + groupPath
+        + "/" + artifact.getArtifactId()
+        + "/maven-metadata.xml";
+  }
+
+  private static class LatestMetadata {
+
+    String latestVersion, lastUpdated;
+
+    LatestMetadata(String latestVersion, String lastUpdated) {
+      this.latestVersion = latestVersion;
+      this.lastUpdated = lastUpdated;
+    }
+  }
+
+  /**
+   * @return null if there was an error in finding the POM file
+   */
+  private static String getSharedDependenciesVersionFromUrl(String pomUrl) {
+    Preconditions.checkNotNull(pomUrl);
+    try {
+      File pomFile = File.createTempFile("pomFile", ".xml");
+      pomFile.deleteOnExit();
+      URL url = new URL(pomUrl);
+      ByteStreams.copy(url.openStream(), new FileOutputStream(pomFile));
+      MavenXpp3Reader read = new MavenXpp3Reader();
+      Model model = read.read(new FileInputStream(pomFile));
+      if (model.getDependencyManagement() == null) {
+        return null;
+      }
+      for (Dependency dep : model.getDependencyManagement()
+          .getDependencies()) {
+        if ("com.google.cloud".equals(dep.getGroupId()) && "google-cloud-shared-dependencies"
+            .equals(dep.getArtifactId())) {
+          return dep.getVersion();
+        }
+      }
+      // No version of google-cloud-shared-dependencies is found within the POM
+      return "";
```

Comment:
why empty string here and null on line 232?

Reply:
I've updated the comments on this method to be more verbose this time around, which I think should properly explain:
   * @return version of google-cloud-shared-dependencies in POM file
   * @return empty string if google-cloud-shared-dependencies was not in the POM file
   * @return null if the POM file could not be found